app/controllers/chat_controller.rb

```diff
@@ -117,8 +117,23 @@ def create_message
     @user_chat_channel_membership.update(last_read_message_id: chat_message_creator.chat_message.id)
 
     if @chat_channel.direct_message_channel?
-      @chat_channel.user_chat_channel_memberships.update_all(following: true)
-      ChatPublisher.publish_new_channel(@chat_channel, @chat_channel.chatable.users)
+
+      # If any of the channel users is ignoring, muting, or preventing DMs from
+      # the current user then we shold not auto-follow the channel once again or
+      # publish the new channel.
+      user_ids_allowing_communication = UserCommScreener.new(
+        acting_user: current_user,
+        target_user_ids: @chat_channel.user_chat_channel_memberships.map(&:user_id)
```

Comment:
Is there any reason to prefer map over pluck here?

Reply:
Oh yeah good point will fix